__fixtures__/test-project/api/src/lib/auth.ts

```diff
@@ -55,7 +55,7 @@ export const hasRole = (roles: AllowedRoles): boolean => {
     return false
   }
 
-  const currentUserRoles = context.currentUser?.roles as string | string[]
+  const currentUserRoles = context?.currentUser?.roles
```

Comment:
This will cause tsc errors in the smoke tests - this is because we ship with code that let's you deal with either form of roles - whether string or array.

See this line here https://github.com/redwoodjs/redwood/blob/9f8362803a21e3c7cfc103cfa1550b2f5ff3d7d9/tasks/test-project/tasks.js#L450 - It probably just doesn't match the original string `context.currentUser?.roles` - without the `context?`. Once you revert that change it should all be ok

<img width="920" alt="Screenshot 2022-08-08 at 11 33 23" src="https://user-images.githubusercontent.com/1521877/183399146-49749a8d-f01f-4c03-9346-e6aae2fabb7e.png">

Reply:
From which file is the screenshot? Cannot reproduce this – not getting any TS error in my live project (either way), but getting this kind of squiggle in the fixture (`__fixtures__/test-project/api/src/lib/auth.ts`):

![grafik](https://user-images.githubusercontent.com/1634615/183513688-be836692-1998-4a44-a307-07a4bffa42ca.png)

either way: 

![grafik](https://user-images.githubusercontent.com/1634615/183513826-db834615-b5cd-43c2-8b71-a50d78a4a965.png)

But that may be simply due to the fact that it's a "dumb" fixture, that's not living in the spot where it would finally sit.

I did some fixups now, have a look. If you insist to take out the optional chaining operator for `context`, i can still easily change that. But in theory this should help and not hurt, but let's see what the CI has to say.